### PR TITLE
[JBPM-9346] Exception when deploying WIH Kafka in wildfly/EAP

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**:  [JBPM-9346](https://issues.redhat.com/browse/JBPM-9346)

### CODE REVIEW

**Comment by gmunozfe** [comment]
@elguardian please review, though basically all credits are to you :)

File: kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java [hunk 1]
```diff
@@ -87,7 +87,11 @@ public KafkaWorkItemHandler(String bootstrapServers,
                    keySerializerClass);
         config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
                    valueSerializerClass);
+        // it is needed to change the classloader to KIEURLClassLoader for dependencies to be resolved and then, set it back
+        ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(KafkaProducer.class.getClassLoader());
         producer = new KafkaProducer<Long, String>(config);
+        Thread.currentThread().setContextClassLoader(oldClassLoader);
```
Comment (elguardian) [inline kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java:94]: try - finally block The oldClassLoader cannot be hanging in there.

Comment (gmunozfe) [reply kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java]: right, done!

**Review by elguardian — APPROVED** [review]
LGTM

**Pull request merged**